### ISSUE TITLE
Cannot use become_user without become. How about it's defined at the upper level ?

### ISSUE THREAD

**Issue by tristanbes** [willthames/ansible-lint#215]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
2.1.2.0 -> ansible
3.4.0 -> ansibn
```

# Desired Behaviour

It's very common to have a playbook that needs to run as a specific user, but at some point, in the role, for specific actions, you need to be root.

When you have `become` defined at the playbook level, and then, `become_user` at the role entry, it shouldn't throw an error:

```
- hosts: web
  become: yes
  become_user: my_user
  roles:
      - { role: role_a }
```

```
- name: Copy the nginx yproxs site configuration file
  become_user: root
  template:
    src: nginx.conf.j2
    dest: /etc/nginx/sites-available/nginx.conf
  tags: vhost
```

# Actual Behaviour (Bug report only)

In the example above, it throws this error:

`[ANSIBLE0017] become_user requires become to work as expected
/home/travis/build/.../tasks/main.yml:8`

In this commit, you cant use only `become_user` without `become` even if it's defined at another level
https://github.com/willthames/ansible-lint/commit/938c2948e66064bcc65719143ffa0dd3feb03281

**Comment by willthames** [comment]
I understand your point, but I find the danger of the false positive outweighs the false negative (that is you could add `become: yes` to your task to pass this test).

However, if this is a deal breaker, I can consider making ANSIBLE0017 an optional task

**Comment by tristanbes** [comment]
fair point. :-)

**Comment by erlangparasu** [comment]
I got error: 
become_user requires become to work as expected ansible-lint partial-become

@willthames Thanks, `become: true` solve it

```yaml
---
- name: Example
  become: true       # <--- (1/2)
  hosts: host_all
  tasks:
    - name: Check installed php8.1-cli
      become_user: root       # <--- (2/2)
      ansible.builtin.apt:
        name: php8.1-cli
        state: present
```
